platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -915,7 +917,9 @@ export default class HangingProtocolService extends PubSubService {
       return protocol.stages.findIndex(it => it.id === stageId);
     }
     if (stageIndex !== undefined) return stageIndex;
-    return 0;
+    // Undefined is returned so that the apply can search for a suitable
+    // stage when initial stages are disabled.
+    return undefined;
```

Comment:
Can we not just replace lines 919-922 with `return stageIndex;` and perhaps your comment too?